node/src/store/in_memory_store.rs

```diff
@@ -118,6 +133,57 @@ impl InMemoryStore {
             .cloned()
             .ok_or(StoreError::LostHash(hash))
     }
+
+    fn mark_header_sampled(&self, height: u64, accepted: bool, cids: Vec<Cid>) -> Result<u64> {
+        if !self.contains_height(height) {
+            return Err(StoreError::NotFound);
+        }
+
+        let metadata = SamplingMetadata {
+            accepted,
+            cids_sampled: cids,
+        };
+
+        if self.sampling_data.insert(height, metadata).is_some() {
+            info!("Overriding existing sampling metadata for height {height}");
+            // modified header wasn't new, no need to update the height
+            Ok(self.get_next_unsampled_height())
+        } else {
+            self.update_cached_sampling_height()
+        }
+    }
+
+    fn update_cached_sampling_height(&self) -> Result<u64> {
```

Comment:
```suggestion
    fn update_lowest_unsampled_height(&self) -> Result<u64> {
```
since it's in memory then everything is 'cached', I think this name reads better for me

Reply:
Agree, changed 👌